src/JsonAssetEnqueuer.php

```diff
@@ -10,97 +12,308 @@ class JsonAssetEnqueuer extends AbstractAssetEnqueuer
+
+        foreach ($jsToRegister as $jsAsset) {
+            $this->wordpressAssetGateway->registerScript($jsAsset['handle'], $jsAsset['src'], $jsAsset['deps'], $jsAsset['ver'], $jsAsset['in_footer']);
         }
     }
 
-    protected function computeDependencyArray($groupName, $availableGroups)
+    /**
+     * @param string $groupName
+     * @param string $dependencyType
+     * @return string[]
+     */
+    protected function computeDependencyArray(string $groupName, string $dependencyType): array
+    {
+        switch ($dependencyType) {
+            case 'js':
+                return $this->manifest->jsDependencies->{$groupName} ?? [];
+            case 'css':
+                return $this->manifest->cssDependencies->{$groupName} ?? [];
+            default:
+                return [];
+        }
+    }
+
+    /** @inheritdoc */
+    public function enqueueStyleGroup(string $groupName)
     {
-        $dependencyArray = isset($this->manifest->jsDependencies->{$groupName}) ? $this->manifest->jsDependencies->{$groupName} : [];
+        $this->wordpressAssetGateway->enqueueStyle($groupName);
 
-        return $dependencyArray;
+        return $this;
     }
 
     /** @inheritdoc */
-    public function enqueueCritical(array $groupNames)
-    {
-        $versionNum = $this->getVersion();
-        foreach ($groupNames as $group) {
-            if (property_exists($this->manifest->js, $group)) {
-                $src = $_SERVER['DOCUMENT_ROOT'] . str_replace($this->container['wwp.asset.folder.prefix'], '', $this->manifest->site->assets_dest) . '/js/' . $group . $versionNum . '.js';
-                if (file_exists($src)) {
-                    $content = apply_filters('wwp.enqueur.critical.js.content', file_get_contents($src));
-                    if (!empty($content)) {
-                        echo '<script id="critical-js">
-                                ' . $content . '
-                            </script>';
-                    }
-                }
+    public function enqueueScriptGroup(string $groupName)
+    {
+        $this->wordpressAssetGateway->enqueueScript($groupName);
+
+        return $this;
+    }
+
+    /** @inheritDoc */
+    public function enqueueStyle(string $handle)
+    {
+        $asset = $this->assetManager->getDependency('css', $handle);
+
+        if ($asset) {
+            $this->enqueueStyleGroup($asset->concatGroup);
+        }
+
+        return $this;
+    }
+
+    /** @inheritDoc */
+    public function enqueueScript(string $handle)
+    {
+        $asset = $this->assetManager->getDependency('js', $handle);
+
+        if ($asset) {
+            $this->enqueueScriptGroup($asset->concatGroup);
+        }
+
+        return $this;
+    }
+
+    /** @inheritDoc */
+    public function inlineStyle(string $handle)
+    {
+        $asset = $this->assetManager->getDependency('css', $handle);
+
+        if ($asset) {
```

Comment:
//This enqueur works with groups, not individual files, hence retrieving the group from the file first, then inlining the group file.```suggestion
        if ($asset) {
```